funsor/terms.py

```diff
@@ -355,6 +355,12 @@ def dtype(self):
     def shape(self):
         return self.output.shape
 
+    def __copy__(self):
+        return self
+
+    def __reduce__(self):
+        return type(self).__origin__, self._ast_values
```

Comment:
Does this mean a pickled lazy expression may be unpickled and evaluated eagerly?

Reply:
That's right, unpickling will respect under the load-time interpretation, not the dump-time interpretation. While I think the dump-time interpretation is slightly more intuitive, I don't see an easy way to implement that (the naive way would invoke O(size)-many reflect handlers). Thus I think it's fine to require users to load under `reflect`. Maybe there is even a use case for eagerly loading an approximation (say a sketch).